envoy/upstream/cluster_manager.h

```diff
@@ -248,28 +254,32 @@ class ClusterManager {
              warming_clusters_.find(cluster) != warming_clusters_.end();
     }
 
-    ClusterConstOptRef getCluster(absl::string_view cluster) {
+    ClusterConstOptRef getCluster(absl::string_view cluster) const {
       auto active_cluster = active_clusters_.find(cluster);
-      if (active_cluster != active_clusters_.end()) {
+      if (active_cluster != active_clusters_.cend()) {
         return active_cluster->second;
       }
       auto warming_cluster = warming_clusters_.find(cluster);
-      if (warming_cluster != warming_clusters_.end()) {
+      if (warming_cluster != warming_clusters_.cend()) {
         return warming_cluster->second;
       }
       return absl::nullopt;
     }
 
     ClusterInfoMap active_clusters_;
     ClusterInfoMap warming_clusters_;
+
+    // Number of clusters that were dynamically added via API (xDS). This will be
+    // less than or equal to the number of `active_clusters_` and `warming_clusters_`.
```

Comment:
We could add some `ASSERT`s here as well.

Reply:
Done.